capstone-frontend/src/components/LeftSideBar.jsx

```diff
@@ -1,44 +1,152 @@
-        <Link to="/" className="bg-dark list-group-item list-group-item-action">
-          <div className="d-flex w-100 justify-content-start align-items-center">
-            <span className="fa fa-tasks fa-fw mr-3"></span>
-            <span id="home-link" className="menu-collapsed"> Home </span>
-          </div>
-        </Link>
-        <Link to="/browse" className="bg-dark list-group-item list-group-item-action">
-          <div className="d-flex w-100 justify-content-start align-items-center">
-            <span className="fa fa-tasks fa-fw mr-3"></span>
-            <span id="browse-link" className="menu-collapsed"> Browse </span>
-          </div>
-        </Link>
-        <Link to="/myreads" className="bg-dark list-group-item list-group-item-action">
-          <div className="d-flex w-100 justify-content-start align-items-center">
-            <span className="fa fa-tasks fa-fw mr-3"></span>
-            <span id="mybooks-link" className="menu-collapsed"> My Books </span>
-          </div>
-        </Link>
-        <Link to="/mylists" className="bg-dark list-group-item list-group-item-action">
-          <div className="d-flex w-100 justify-content-start align-items-center">
-            <span className="fa fa-tasks fa-fw mr-3"></span>
-            <span id="mylists-link" className="menu-collapsed"> My Lists </span>
-          </div>
-        </Link>
-        <Link to="/myclubs" className="bg-dark list-group-item list-group-item-action">
-          <div className="d-flex w-100 justify-content-start align-items-center">
-            <span className="fa fa-tasks fa-fw mr-3"></span>
-            <span id="myclubs-link" className="menu-collapsed"> My Clubs </span>
-          </div>
-        </Link>        
-      </ul>
-    </div>
-  );
-};
-
-export default LeftSideBar;
\ No newline at end of file
+import { Accordion, Card, Button } from 'react-bootstrap';
+import "../styles/LeftSideBar.css";
+
+
+export class LeftSideBar extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      bookLists: []
+    }
+  }
+
+  fetchBookLists = async () => {
+
+    const userID = window.localStorage.getItem("userID");
+
+    const bookLists = await fetch(`/api/booklist?userID=${userID}`, {
+      method: "GET",
+    }).then(resp => resp.json());
+
+    this.setState({ bookLists });
+  }
+
+  componentDidMount() {
+    this.fetchBookLists();
+  }
+
+  deleteBookList = async (bookListId) => {
+
+    // Delete BookList in Firebase
+    await fetch(`/api/booklist?id=${bookListId}`, {
+      method: "DELETE",
+    });
+
+    this.fetchBookLists();
+
+    // These lines of code check if the current BookList URL has the same id as the one being deleted
+    // If so we redirect home
+    const currentUrlPath = new URL(window.location.href).pathname;
+    const bookListIdURL = currentUrlPath.substr(10);
+
+    if (bookListIdURL === bookListId) {
+      document.location.href = "/";
+    }
```

Comment:
Oh I see! Is it possible to do something like currentUrlPath.includes(bookListId) instead of this?

Reply:
Fix